src/armature/Generator.ts

```diff
@@ -417,9 +504,13 @@ export class Generator {
         const finalInstance = <GeneratorInstance>minBy(instances, (instance: GeneratorInstance) =>
             instance.getCost()
         );
-        range(0, Math.max(0, finalDepth - sosmcDepth)).forEach(() =>
-            finalInstance.growIfPossible()
-        );
+        while (finalInstance.getSpawnPoints().length > 0) {
```

Comment:
nit: this is pretty similar to the last bit in `generate`. Would it be possible to make a helper function to do this?

Reply:
I think the difference in scope makes it a bit difficult to make a helper function for both of these cases. I do see the overlap though. I'd like to put a TODO to defer to later, or maybe to somehow who has a better handle of how the scoping works here.